src/api/cpp/cvc5_checks.h

```diff
@@ -164,76 +164,78 @@ namespace cvc5 {
                 << (idx) << ", expected "
 
 /* -------------------------------------------------------------------------- */
-/* Solver checks.                                                             */
+/* Node manager check.                                                        */
```

Comment:
I don't quite follow why the style of these checks changed, e.g. why we need to pass a `nm` as an explicit argument instead of relying on `this->`?

Shouldn't this be analogous?

Reply:
Actually, I think I see the issue, perhaps some but not all classes have been converted Solver -> NodeManager.  Meaning that we need to extract node manager from solver on the ones that haven't converted, e.g. for `Grammar`.

However, I would be hopeful here that consistently *all* classes make this conversion, including `Grammar`, after which maybe the extra arguments to the checks are not necessary?

Reply:
Right, so, Grammar was the reason. I was not sure if we want to keep the Solver reference, or if we also want to convert this to NM. I guess it is the latter then?

Reply:
I'll update Grammar.